Configure

```diff
@@ -919,8 +919,17 @@ $target{dso_extension}=$target{shared_extension_simple};
 $config{cross_compile_prefix} = $ENV{'CROSS_COMPILE'}
     if $config{cross_compile_prefix} eq "";
 
+# This is textbook from the perlvar manual.  It makes sure we have
+# a perl path that can be used as a file name, or as the case may be,
+# as something that can be used with the hashbang script syntax.
+use Config;
+my $secure_perl_path = $Config{perlpath};
```

Comment:
It totally depends on how we use it.  Right now, we use it to hack installed perl scripts, and the result might become `#!perl`, as pointed out in #1199.  This is actually warning against in `perlvar`:

```
               It is not safe to use the value of $^X as a path name of a
               file, as some operating systems that have a mandatory suffix on
               executable files do not require use of the suffix when invoking
               a command.  To convert the value of $^X to a path name, use the
               following statements:

                   # Build up a set of file names (not command names).
                   use Config;
                   my $this_perl = $^X;
                   if ($^O ne 'VMS') {
                       $this_perl .= $Config{_exe}
                         unless $this_perl =~ m/$Config{_exe}$/i;
                       }

               Because many operating systems permit anyone with read access
               to the Perl program file to make a copy of it, patch the copy,
               and then execute the copy, the security-conscious Perl
               programmer should take care to invoke the installed copy of
               perl, not the copy referenced by $^X.  The following statements
               accomplish this goal, and produce a pathname that can be
               invoked as a command or referenced as a file.

                   use Config;
                   my $secure_perl_path = $Config{perlpath};
                   if ($^O ne 'VMS') {
                       $secure_perl_path .= $Config{_exe}
                           unless $secure_perl_path =~ m/$Config{_exe}$/i;
                       }

```

However, if we take one of the choices from #1202, this PR becomes entirely obsolete.